### PR TITLE
installer: Remove dead code

### PR DESCRIPTION
**Pull request by Quentin-M**
Following our documented process of migrating out of the CloudFormation and Go-powered bare-metal provisioning methods, towards using exclusively TerraForm, this PR aims at removing all the code that is exclusively used for the provisioning - and refactoring/consolidating the rest.

### CODE REVIEW

**Comment by sym3tri** [comment]
@Quentin-M how is this going? we should delete things sooner ASAP. The old code is causing confusion.

**Comment by Quentin-M** [comment]
@sym3tri I pretty much stopped working on it for now after @ggreer told me to hold off removing everything for now. Still pushed the first commit as a PR though to show that it's being done and to save the progress.

**Comment by Quentin-M** [comment]
I have a LOT coming into that PR. I will update it as soon as I get time to finish what I started locally.

**Comment by Quentin-M** [comment]
This PR appears to be working fine.

I see a weird behavior on the UI with the advanced networking and I'd like to ask what's going on here. It might actually be on master. I need to verify. There is no real error, and the backend seems to be answering fine (there was no real modification here anyways) but looks like JS is not filling some parameters to the backend.. Otherwise, all good.

No need to look at the commits themselves, there is simply no way 👎  
You can however take a look at the directory layout if you were familiar with the backend!

**Comment by s-urbaniak** [comment]
> No need to look at the commits themselves, there is simply no way 👎

well ... LGTM ... it's fine ;-)

File: installer/Makefile [hunk 1]
```diff
@@ -2,30 +2,30 @@ export CGO_ENABLED:=0
...
 	GOOS=$* go build -o bin/$*/installer -ldflags $(LD_FLAGS) $(REPO)/cmd/installer
 
 .PHONY: backend
-backend: binassets/assets.go
+backend: assets/bindata.go
 	GOOS=$(OS) go build -o bin/$(OS)/installer -ldflags $(LD_FLAGS) $(REPO)/cmd/installer
 
-binassets/assets.go: frontend bin/go-bindata $(shell find assets -type f)
-	./bin/go-bindata -pkg binassets -o binassets/assets.go -prefix assets assets/...
+assets/bindata.go: frontend bin/go-bindata $(shell find assets -type f)
```
Comment (ggreer) [inline installer/Makefile]: This creates a circular dependency. `assets/bindata.go` is in `assets/`

File: installer/frontend/__tests__/protocols.js [hunk 2]
```diff
@@ -1,90 +0,0 @@
-/* eslint-env jest, node */
```
Comment (ggreer) [inline installer/frontend/__tests__/protocols.js]: Why did you delete this test? We were using this!

File: installer/examples/tectonic-aws.progress [hunk 3]
```diff
@@ -1,57 +0,0 @@
-{
```
Comment (ggreer) [inline installer/examples/tectonic-aws.progress]: All of these example progress files were used by our front end tests.

**Comment by ggreer** [comment]
This PR should not have been merged. It broke a lot of things.

File: installer/frontend/__tests__/protocols.js (hunk 2 above)
Comment (Quentin-M) [inline installer/frontend/__tests__/protocols.js]: I thought this was only for comparing with the JSON examples to the backend's create endpoint, which we don't use anymore.

File: installer/Makefile (hunk 1 above)
Comment (Quentin-M) [inline installer/Makefile]: I see, I can follow up to grep this one out.

File: installer/examples/aws-vpc.json [hunk 4]
```diff
@@ -1,54 +0,0 @@
-{
```
Comment (ggreer) [inline installer/examples/aws-vpc.json]: Just like the progress files, we needed these for the front end tests.

File: installer/Makefile [hunk 5]
```diff
@@ -2,30 +2,30 @@ export CGO_ENABLED:=0
...
+	@./bin/go-bindata -pkg assets -o assets/bindata.go -ignore=bindata.go -ignore=doc.go -ignore=assets.go -prefix assets assets/...
```
Comment (ggreer) [inline installer/Makefile]: Prefixing with an `@` doesn't echo the command, making it harder to debug what's going on.

If doc.go & assets.go are ignored by go-bindata, shouldn't they also be removed from Makefile dependencies? Right now, `shell find …` will catch them.

**Comment by Quentin-M** [comment]
Followed up with https://github.com/coreos/tectonic-installer/pull/655

**Comment by Quentin-M** [comment]
Followed up with https://github.com/coreos/tectonic-installer/pull/655

**Comment by dghubble** [comment]
Did you check with @ethernetdan? I suspect this breaks his bare-metal testing efforts.

**Comment by Quentin-M** [comment]
I asked him IRL last week.

**Comment by dghubble** [comment]
Thanks

File: installer/api/api.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+	mux.Handle("/aws/vpcs", httpHandler("POST", ctx, awsGetVPCsHandler))
+	mux.Handle("/aws/vpcs/subnets", httpHandler("POST", ctx, awsGetVPCsSubnetsHandler))
+	mux.Handle("/aws/ssh-key-pairs", httpHandler("POST", ctx, awsGetKeyPairsHandler))
+	mux.Handle("/aws/zones", httpHandler("POST", ctx, awsGetZonesHandler))
+	mux.Handle("/aws/domain", httpHandler("POST", ctx, awsGetDomainInfoHandler))
+
+	// handlers_terraform.go
+	mux.Handle("/terraform/apply", httpHandler("POST", ctx, terraformApplyHandler))
+	mux.Handle("/terraform/status", httpHandler("POST", ctx, terraformStatusHandler))
+	mux.Handle("/terraform/assets", httpHandler("POST", ctx, terraformAssetsHandler))
```
Comment (ggreer) [inline installer/api/api.go]: This broke downloading assets. The HTTP method for that is GET, not POST. Fixed in https://github.com/coreos/tectonic-installer/pull/674
